compiler/types/TyCoRep.hs

```diff
@@ -2526,6 +2522,35 @@ substForAllCoBndrUsing sym sco (TCvSubst in_scope tenv cenv)
 
     new_var  = uniqAway in_scope (setTyVarKind old_var new_ki1)
 
+substForAllCoCoVarBndrUsing :: Bool  -- apply sym to binder?
+                            -> (Coercion -> Coercion)  -- transformation to kind co
+                            -> TCvSubst -> CoVar -> KindCoercion
+                            -> (TCvSubst, CoVar, KindCoercion)
+substForAllCoCoVarBndrUsing sym sco (TCvSubst in_scope tenv cenv)
+                            old_var old_kind_co
+  = ( TCvSubst (in_scope `extendInScopeSet` new_var) tenv new_cenv
+    , new_var, new_kind_co )
+  where
+    new_cenv | no_change && not sym = delVarEnv cenv old_var
+             | otherwise = extendVarEnv cenv old_var new_co
+
+    new_co = mkCoVarCo new_var
+    no_kind_change = noFreeVarsOfCo old_kind_co
+    no_change = no_kind_change && (new_var == old_var)
+
+    new_kind_co | no_kind_change = old_kind_co
+                | otherwise      = sco old_kind_co
+
+    -- TODO: unsure
```

Comment:
So am I. (Unsure, that is) It would be helpful to illustrate this with an example, connecting the pieces of the example back to the variables here.